cg/meta/workflow/taxprofiler.py

```diff
@@ -18,3 +34,82 @@ def __init__(
         pipeline: Pipeline = Pipeline.TAXPROFILER,
     ):
         super().__init__(config=config, pipeline=pipeline)
+        self.root_dir: str = config.taxprofiler.root
+
+    @property
+    def root(self) -> str:
+        return self.root_dir
+
+    @property
+    def fastq_handler(self):
+        return TaxprofilerFastqHandler
+
+    def get_case_config_path(self, case_id):
+        return NextflowAnalysisAPI.get_case_config_path(case_id=case_id, root_dir=self.root_dir)
+
+    @staticmethod
+    def build_sample_sheet_content(
+        case_id: str,
+        fastq_r1: List[str],
+        fastq_r2: List[str],
+        instrument_platform: SequencingPlatform,
+        fasta: Optional[str] = "",
+    ) -> Dict[str, List[str]]:
+        """Build sample sheet headers and lists."""
+        try:
+            TaxprofilerSample(
+                sample=case_id,
+                fastq_r1=fastq_r1,
+                fastq_r2=fastq_r2,
+                instrument_platform=instrument_platform,
+            )
+        except ValidationError as error:
+            LOG.error(error)
+            raise ValueError
+
+        # Complete sample lists to the same length as fastq_r1:
+        samples_full_list: List[str] = [case_id] * len(fastq_r1)
+        instrument_full_list: List[str] = [instrument_platform] * len(fastq_r1)
+        fasta_full_list: List[str] = [fasta] * len(fastq_r1)
+
+        samplesheet_content: Dict[str, List[str]] = {
+            NFX_SAMPLE_HEADER: samples_full_list,
+            TAXPROFILER_RUN_ACCESSION: samples_full_list,
+            TAXPROFILER_INSTRUMENT_PLATFORM: instrument_full_list,
+            NFX_READ1_HEADER: fastq_r1,
+            NFX_READ2_HEADER: fastq_r2,
+            TAXPROFILER_FASTA_HEADER: fasta_full_list,
+        }
+        return samplesheet_content
+
+    def write_samplesheet(
+        self, case_id: str, instrument_platform: SequencingPlatform, fasta: Optional[str]
+    ) -> None:
+        """Write sample sheet for taxprofiler analysis in case folder."""
+        case: Family = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        for link in case.links:
+            sample_metadata: List[str] = self.gather_file_metadata_for_sample(link.sample)
+            fastq_r1: List[str] = NextflowAnalysisAPI.extract_read_files(1, sample_metadata)
+            fastq_r2: List[str] = NextflowAnalysisAPI.extract_read_files(2, sample_metadata)
+            samplesheet_content: Dict[str, List[str]] = self.build_sample_sheet_content(
+                case_id, fastq_r1, fastq_r2, instrument_platform, fasta
+            )
+            LOG.info(samplesheet_content)
+            NextflowAnalysisAPI.create_samplesheet_csv(
```

Comment:
This function should be
```suggestion
            NextflowAnalysisAPI.create_sample_sheet(
```
And use the io/controller for writing csv files, but in a future PR

Reply:
Since this function is a common one for both rnafusion and taxprofiler, would that be ok to change the name later when we would do code refactoring?